File: 프로그래머스/lv2/42746. 가장 큰 수/가장 큰 수.py
def solution(numbers):
    numbers = list(map(str, numbers))
    numbers.sort(key=lambda x: (x * 4)[:4], reverse=True)
    
    
    # 테케 11번 아마 0이 원소인 경우임 -> 000이 아닌 0 리턴하도록
    if numbers[0] == '0':
        return '0'
    
    return ''.join(numbers)

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;

class Solution {
    public String solution(int[] numbers) {
        
        String[] strNumbers = new String[numbers.length];
        
        for (int i=0; i < numbers.length; i++) {
            strNumbers[i] = String.valueOf(numbers[i]);
        }
        
        // 두 문자열을 붙였을 때 더 큰 순으로 정렬하기
        Arrays.sort(strNumbers, (a, b) -> (b + a).compareTo(a + b));
        
        if (strNumbers[0].equals("0")) {
            return "0";
        }
        
        StringBuilder answer = new StringBuilder();
        
        for (String num : strNumbers) {
            answer.append(num);
        }
        
        return answer.toString();
    }
}